### ISSUE TITLE
Instructions for building on windows

### ISSUE THREAD

**Issue by ahxm** [openzipkin/zipkin#1501]
hello,
  I can run zipkin-server locally,but I cannot access http://localhost:9411/ ,the response is :There was an unexpected error (type=Internal Server Error, status=500).class path resource [zipkin-ui/index.html] cannot be resolved to URL because it does not exist。I don't know  how to start zipkin-ui at windows.

**Comment by codefromthecrypt** [comment]
@ahxm do you have any errors from the console? I assume you are running `java jar zipkin-server.jar` as noted in the README?

@smartthought @gena01 @Horusiath @theoptimiste I think each of you have run zipkin locally. Did you have to do anything special in order to do so? We are sorely missing a windows section on readme.

@eirslett @abesto maybe we should setup a CI worker to use free windows VM to test zipkin can be started on windows (if not also that it can be built)

**Comment by ahxm** [comment]
I think it is really necessary that detailed steps  to run zipkin in windows environment.
I am not running  java jar zipkin-server.jar 
I use IntelliJ IDEA to run ZipkinServer.main(). 
localhost:9411/config.json is OK
localhost:9411/health is OK
localhost:9411/info is OK
localhost:9411/metrics is OK
localhost:9411/ is error
there is the errors from the console:
2017-01-23 11:05:32.000 ERROR 4712 --- [io-9411-exec-10] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet dispatcherServlet threw exception

java.io.FileNotFoundException: class path resource [zipkin-ui/index.html] cannot be resolved to URL because it does not exist
	at org.springframework.core.io.ClassPathResource.getURL(ClassPathResource.java:187) ~[spring-core-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.core.io.AbstractFileResolvingResource.contentLength(AbstractFileResolvingResource.java:151) ~[spring-core-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.http.converter.ResourceHttpMessageConverter.getContentLength(ResourceHttpMessageConverter.java:99) ~[spring-web-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.http.converter.ResourceHttpMessageConverter.getContentLength(ResourceHttpMessageConverter.java:50) ~[spring-web-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.http.converter.AbstractHttpMessageConverter.addDefaultHeaders(AbstractHttpMessageConverter.java:260) ~[spring-web-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.http.converter.AbstractHttpMessageConverter.write(AbstractHttpMessageConverter.java:205) ~[spring-web-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodProcessor.writeWithMessageConverters(AbstractMessageConverterMethodProcessor.java:247) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.HttpEntityMethodProcessor.handleReturnValue(HttpEntityMethodProcessor.java:203) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.method.support.HandlerMethodReturnValueHandlerComposite.handleReturnValue(HandlerMethodReturnValueHandlerComposite.java:81) ~[spring-web-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:132) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85) ~[spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963) [spring-webmvc-4.3.5.RELEASE.jar:4.3.5.RELEASE]
	at 
then I  query the readme document in zipkin-ui,there is a step:
In another terminal, launch the zipkin UI server:

# Do this in another terminal!

$ cd zipkin-ui

$ proxy=http://localhost:9411 ./npm.sh run dev

but I don't know how to execute "proxy=http://localhost:9411 ./npm.sh run dev"  command at windows. I found a lot of people have run zipkin local success,I am not sure what environment they did. Maybe they all run in a Linux environment

**Comment by codefromthecrypt** [comment]
FWIW the zipkin-ui readme does not suggest to run zipkin server in
intellij. Did you try what it suggests?

https://github.com/openzipkin/zipkin/tree/master/zipkin-ui#whats-the-easiest-way-to-develop-against-this-locally

It would be helpful to figure out what works at all for you.. for example,
if you failed to use the normal instructions and then used intellij for
some reason..

**Comment by abesto** [comment]
I'm wondering if any @openzipkin/devops-tooling members have Windows experience? I have only the basics - could probably set something up with https://www.appveyor.com/, but it won't be pretty.

**Comment by ahxm** [comment]
@adriancole  I run zipkin  java jar zipkin-server.jar as noted in the README in linux centos6.5  is OK
I access http://localhost:9411/  no problems. I think you all dev invironment is linux ,so it seems  rare to find Windows environment development experience about zipkinO(∩_∩)O

**Comment by codefromthecrypt** [comment]
per #1532 folks who are trying to build on windows may need to avoid building the zipkin-ui project.

ex. `mvnw -DskipTests --also-make -pl -:zipkin-ui,zipkin-server clean install`

**Comment by zxeoc** [comment]
To people who got the same issue, I think the reason was that @ahxm  forgot to run mvn compile in intellij. And don't forgot to make sure npm install (which include in the mvn compile) was successful.

**Comment by jorgheymans** [comment]
The instructions for building now work on windows as well : https://github.com/openzipkin/zipkin#running-the-server-from-source

**Comment by swiatkol** [comment]
I had a similar issue with Ubuntu. zipkin-lens/index.html was not present in compiler resources.
I tried `./mvnw -DskipTests --also-make -pl zipkin-server clean install` and then run ZipkinServer from IntelliJ but it didn't work.
I needed to clear the cache by removing `~/.cache/JetBrains/IntelliJIdea2020.3` folder (the "Invalidate cache and restart" option didn't work either).
I know it wasn't on Windows but hope it helps.
